src/controllers/ContactController.php

```diff
@@ -56,23 +96,19 @@ public function contact(Request $request, PDO $db)
         }
 
         // run it by akismet if we have it
-        if (isset($this->config['akismet']['apiKey'], $this->config['akismet']['blog'])) {
-            $spamCheckService = new SpamCheckService(
-                $this->config['akismet']['apiKey'],
-                $this->config['akismet']['blog']
-            );
-            $isValid          = $spamCheckService->isCommentAcceptable(
-                $data['comment'],
+        if (isset($this->spamCheckService)) {
```

Comment:
Can we enforce this dependency instead, e.g. with a SpamCheckInterface and a dummy class that always returns true? That'll simplify code paths here.

Reply:
Added a NullSpamCheckService and a SpamCheckServiceInterface.
Good call!!